doltpy/cli/write/write.py

```diff
@@ -0,0 +1,172 @@
+from typing import List, Callable, Mapping, Any
+import pandas as pd
+from doltpy.cli import Dolt
+from doltpy.shared.helpers import columns_to_rows
+import logging
+import io
+import tempfile
+import os
+import csv
+from datetime import datetime
+
+logger = logging.getLogger(__name__)
+
+CREATE, FORCE_CREATE, REPLACE, UPDATE = 'create', 'force_create', 'replace', 'update'
+IMPORT_MODES_TO_FLAGS = {CREATE: ['-c'],
+                         FORCE_CREATE: ['-f', '-c'],
+                         REPLACE: ['-r'],
+                         UPDATE: ['-u']}
+
+
+def write_file(dolt: Dolt,
+               table: str,
+               file_handle: io.StringIO,
+               # TODO what to do about this?
+               filetype: str = 'csv',
+               import_mode: str = None,
+               primary_key: List[str] = None,
+               commit: bool = False,
+               commit_message: str = None,
+               commit_date: datetime = None):
+    def writer(filepath: str):
+        with open(filepath, 'w') as f:
+            f.writelines(file_handle.readlines())
+
+    _import_helper(dolt, table, writer, primary_key, import_mode, commit, commit_message, commit_date)
+
+
+def write_columns(dolt: Dolt,
+                  table: str,
+                  columns: Mapping[str, List[Any]],
+                  import_mode: str = None,
+                  primary_key: List[str] = None,
+                  commit: bool = True,
+                  commit_message: str = False,
+                  commit_date: datetime = None):
+    """
+
+    :param dolt:
+    :param table:
+    :param columns:
+    :param import_mode:
+    :param primary_key:
+    :param commit:
+    :param commit_message:
+    :param commit_date:
+    :return:
+    """
+    def writer(filepath: str):
+        assert len(list(set(len(col) for col in columns.values()))) == 1, 'Must pass columns of identical length'
+
+        with open(filepath, 'w') as f:
+            csv_writer = csv.DictWriter(f, columns.keys())
+            rows = columns_to_rows(columns)
+            csv_writer.writeheader()
+            csv_writer.writerows(rows)
+
+    _import_helper(dolt, table, writer, primary_key, import_mode, commit, commit_message, commit_date)
+
+
+def write_rows(dolt: Dolt,
```

Comment:
nit re: backwards compatibility: any hesitation about having `dolt.read` and `dolt.write` that multiplex on parameters, or `dolt.write_df`, `dolt.write_dict` and `dolt.write_json` extras for the other standard non-array types that won't change? (there are other types of dataframes also, like pyarrow)

Reply:
`write_rows` is effectively `write_dicts`, it's in reference to the Python data structure. I am open to having `write_json`, etc., but not totally sold because these just dispatch into the CLI anyway.